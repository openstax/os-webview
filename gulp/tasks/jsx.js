const path = require('path');
const gulp = require('gulp');
const config = require('../config');
const pi = require('gulp-load-plugins')({
    pattern: ['gulp-*', 'gulp.*', 'del']
});

function jsx() {
    const configSrcApp = `${config.src}/app`;
    const configDestApp = `${config.dest}/app`;

    return gulp.src(`${configSrcApp}/**/*.jsx`, {
        since: gulp.lastRun(jsx)
    })
    .pipe(pi.sourcemaps.init({loadMaps: true}))
    .pipe(pi.rename((uri) => {
        uri.extname = '.jsx.js';
    }))
    .pipe(pi.babel({
        compact: false,
        presets: ['@babel/preset-react']
    }))
    // prefix the sourcemaps with '../src/' so webpack can find them
    .pipe(pi.sourcemaps.mapSources(function(sourcePath, file) {
      const sourcePathHtml = sourcePath.replace('.jsx.js', '.jsx');
      const rel = path.relative(path.dirname(`${configDestApp}/${sourcePath}`), `${configSrcApp}/${sourcePathHtml}`)
      return rel;
    }))
    .pipe(pi.sourcemaps.write('.'))
    .pipe(gulp.dest(`${configDestApp}`));
}


function watchJsx() {
    gulp.watch([`${config.src}/**/*.jsx`], config.watchOpts, gulp.series(
        jsx
    ));
}

exports.jsx = jsx;
exports.jsx.watch = watchJsx;
