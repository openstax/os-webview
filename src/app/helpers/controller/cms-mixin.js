import $ from '~/helpers/$';
import bookPromise from '~/models/book-titles';
import {urlFromSlug} from '~/models/cmsFetch';
import React, {useState, useEffect} from 'react';
import LoadingPlaceholder from '~/components/loading-placeholder/loading-placeholder';

export function transformData(data) {
    Reflect.ownKeys(data).forEach((prop) => {
        if (Array.isArray(data[prop])) {
            const arr = data[prop];
            const contentItem = arr.filter((e) => e.type === 'content').length === 1;

            if (contentItem) {
                data[prop] = {};
                arr.forEach((v) => {
                    data[prop][v.type] = v.value;
                });
            } else {
                data[prop] = arr.map((item) => {
                    if (item.value) {
                        return item.value;
                    }
                    return item;
                });
            }
        }
    });

    return data;
}

function loadImages(data) {
    const promises = [];

    if (typeof data.image === 'number') {
        promises.push(fetch(`${$.apiOriginAndPrefix}/images/${data.image}`)
            .then((response) => response.json())
            .then((json) => {
                data.image = json.file || (json.meta || {}).download_url;
            }));
    }

    Reflect.ownKeys(data).forEach((prop) => {
        if (typeof data[prop] === 'object' && data[prop] !== null) {
            promises.push(loadImages(data[prop]));
        }
    });

    return Promise.all(promises);
}

async function getUrlFor(initialSlug) {
    let apiUrl = urlFromSlug(initialSlug);

    // A little magic to handle book titles
    if (initialSlug.startsWith('books/')) {
        const strippedSlug = initialSlug.substr(6);

        if (strippedSlug) {
            const bookList = await bookPromise;
            const bookEntry = bookList.find((e) => e.meta.slug === strippedSlug);

            if (bookEntry) {
                apiUrl = bookEntry.meta.detail_url;
            }
        }
    }

    const qsChar = ((/\?/).test(apiUrl)) ? '&' : '?';

    return `${apiUrl}${qsChar}format=json`;
}

export async function fetchFromCMS(slug, preserveWrapping=false) {
    const apiUrl = await getUrlFor(slug);
    let data;

    try {
        data = await (await fetch(apiUrl)).json();
    } catch (err) {
        console.warn(`ERROR fetching slug ${slug}: ${err}`);
        data = {error: err};
    }

    data.slug = slug;
    return preserveWrapping ? data : transformData(data);
}

async function fetchPageData({slug, preserveWrapping, setsPageTitleAndDescription=true}) {
    const data = await fetchFromCMS(slug, preserveWrapping);

    await loadImages(data);
    if (setsPageTitleAndDescription) {
        $.setPageTitleAndDescriptionFromBookData(data);
    }
    return data;
}

export function usePageData(fpdParams) {
    const [pageData, setPageData] = useState();
    let statusPage = null;

    useEffect(() => {
        fetchPageData(fpdParams).then(setPageData);
    }, [fpdParams.slug]); // eslint-disable-line react-hooks/exhaustive-deps

    if (!pageData) {
        statusPage = <LoadingPlaceholder />;
    } else if (pageData.error) {
        statusPage =
            <div className="page error">
                Unable to load page: {pageData.error}
            </div>
        ;
    }

    return [pageData, statusPage];
}

export function useTextFromSlug(slug) {
    const [text, setText] = React.useState();

    React.useEffect(() => {
        const url = urlFromSlug(slug);

        fetch(url).then((r) => r.text()).then((pageHtml) => {
            const parser = new window.DOMParser();
            const newDoc = parser.parseFromString(pageHtml, 'text/html');

            setText(newDoc.body.innerHTML);
        });
    }, [slug]);

    return text;
}
