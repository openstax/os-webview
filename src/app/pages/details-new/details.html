<template args="model">
    <div class="hero" data-slug="{model.slug}">
        <div class="content book-title">
            <h1>{model.bookTitle}</h1>
        </div>
    </div>
    <div class="content-wrapper">
        <div class="content">
            <div class="phone-view" skip="true"></div>
            <div class="bigger-view">
                <div class="tab-controller" skip="true"></div>
                <div class="tab-content" skip="true"></div>
            </div>
        </div>
    </div>
</template>
