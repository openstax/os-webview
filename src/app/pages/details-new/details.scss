@import 'pattern-library/core/pattern-library/headers';

@mixin set-color-items($color) {
    @at-root .#{$color} {
        .hero,
        .book-icon,
        .webinars .icon-cell {
            background-color: os-color($color);
            color: text-color(white);
        }

        .hero {
            background-image: linear-gradient(to right, os-color($color), lighten(os-color($color), 7));
        }

        @if ($color == yellow) {
            .book-icon,
            .webinars .icon-cell {
                color: text-color(normal);
            }
        }
    }

    @at-root .fg-#{$color} {
        .hero {
            color: os-color($color);
        }
    }
}

.details-page-v2 {

    @include wider-than($phone-max) {
        background-color: ui-color(white);

        &.card-background {
            > .content-wrapper {
                background-image: linear-gradient(to bottom, ui-color(white), ui-color(white) 10rem, ui-color(default) 25rem);

                @include wider-than($tablet-max) {
                    background-image: linear-gradient(to bottom, ui-color(white), ui-color(white) 15rem, ui-color(default) 30rem);
                }
            }
        }
    }

    .hero {
        @include set-color-items(gold);
        @include set-color-items(green);
        @include set-color-items(blue);
        @include set-color-items(deep-green);
        @include set-color-items(gray);
        @include set-color-items(light-blue);
        @include set-color-items(orange);
        @include set-color-items(yellow);
        @include set-color-items(red);

        align-items: flex-end;
        background-image: linear-gradient(75deg, #aaa, #777);
        color: text-color(normal);
        display: flex;
        min-height: 16.5rem;
        overflow: hidden;
        padding-bottom: 5rem;

        @include width-between($phone-max, $tablet-max) {
            min-height: 27.3rem;
            padding-bottom: 6rem;
        }

        @include wider-than($tablet-max) {
            min-height: 34.3rem;
            padding-bottom: 7rem;
        }

        .book-title {
            padding: 0;

            h1 {
                @include title-font(4rem);

                text-align: left;
                // white-space: nowrap;

                @include width-between($phone-max, $tablet-max) {
                    @include title-font(10rem);
                }

                @include wider-than($tablet-max) {
                    @include title-font(13rem);
                }
            }
        }
    }

    .content {
        @extend %content;

        justify-content: flex-start;
    }

    .phone-view {
        display: block;

        @include wider-than($phone-max) {
            display: none;
        }
    }

    .bigger-view {
        display: none;

        @include wider-than($phone-max) {
            display: block;
        }
    }

    .tab-controller {
        width: 100%;

        .tab {
            @include set-font(body-large);

            font-weight: normal;

            @include wider-than($tablet-max) {
                font-size: 150%;
                padding-bottom: $normal-margin;
            }
        }
    }

    .tab-content {
        padding: 4rem 0 6rem;
        width: 100%;
    }

    .unbold {
        font-weight: normal;
    }

    .resources {
        display: grid;
        grid-gap: 2 * $normal-margin  2 * $normal-margin;
        margin: 2rem 0 0;

        @include wider-than($phone-max) {
            grid-template-columns: 1fr 1fr;
        }

        @include wider-than($tablet-max) {
            grid-template-columns: 1fr 1fr 1fr;
        }

        &.ally {
            @include wider-than($phone-max) {
                grid-template-columns: 1fr 1fr 1fr;
            }

            @include wider-than($tablet-max) {
                grid-template-columns: 1fr 1fr 1fr 1fr;
            }
        }
    }

    .btn.default {
        background-color: ui-color(white);
    }

    .table-of-contents {
        list-style: none;
        margin: 0;
        padding: 0;
        padding-left: 2rem;

        > li {
            font-weight: bold;
            margin-bottom: 2rem;
        }

        ol {
            font-weight: normal;
            list-style: none;
            padding-left: 2rem;

            .box {
                font-weight: bold;
            }
        }
    }

}
