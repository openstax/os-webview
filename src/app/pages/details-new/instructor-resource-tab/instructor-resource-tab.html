<template args="model">
    <div>
        <!-- TODO: Get this stuff from the CMS -->
        <h2>Free stuff. <span class="unbold">No catch.</span></h2>
        <div class="free-stuff-blurb">
            Access free resources integrated with your book. Sign up for an account
            to access locked content.
        </div>
        <div class="resources">
            <div class="resource-box double">
                <h3>OpenStax Community Resources</h3>
                <div class="blurb">
                    Placeholder text about access, develop, share resources created by the
                    OpenStax Community that align with OpenStax {model.bookTitle}.<br>
                    <a href="http://oercommons.org">
                        Visit the hub
                        <i class="fa fa-external-link-alt"></i>
                    </a>
                </div>
                <div class="bottom">
                    <div class="left"></div>
                    <div class="right">
                        Download Chapter 5 enhanced lecture slides???
                    </div>
                </div>
            </div>
            <resource-box each="resource in model.resources" skip="true"></resource-box>
        </div>
    </div>
    <!-- TODO pull url from CMS? -->
    <a href="/blog/upcoming-openstax-webinars" class="webinars">
        <div class="icon-cell"><span class="fa fa-desktop"></span></div>
        <div>
            <!-- TODO pull text from CMS -->
            <h2>Find a webinar</h2>
            <div class="blurb">
                Learn more by attending
                one of our free webinars featuring
                OpenStax authors, staff, and partners.
            </div>
        </div>
    </a>
    <hr>
    <if condition="model.allies.length">
        <div class="ally-blurb">
            <!-- TODO Get this from the CMS -->
            <h2>Want more?</h2>
            <div class="blurb-body">
                OpenStax Partners provide additional, low-cost resources for your
                book. See what's offered for this title below and learn more about
                our partners <a href="/partners">here</a>.
            </div>
            <div class="ally-logo">
                <img
                src="http://staxlogos.openstax.org/OSX-ALLY-Gray-RGB-150dpi.png"
                alt=""
                >
            </div>
        </div>
        <div class="ally resources">
            <div each="ally in model.allies" class="ally-box resource-box">
                <div class="top">
                    <img src="{ally.ally_color_logo}">
                </div>
                <div class="bottom">
                    <a href="{ally.book_link_url}">
                        {ally.book_link_text}<i class="fa fa-external-link-alt"></i>
                    </a>
                </div>
            </div>
        </div>
    </if>

</template>
