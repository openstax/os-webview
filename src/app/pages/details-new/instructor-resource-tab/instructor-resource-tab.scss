@import 'pattern-library/core/pattern-library/headers';

$icon-size: 6rem;

.instructor-resources {
    display: grid;
    grid-gap: 4rem;
    margin-bottom: 3.5rem;

    h2 {
        margin: 0;
    }

    .free-stuff-blurb {
        display: grid;
        grid-gap: $normal-margin;
        grid-template-columns: auto;
        margin-bottom: 4rem;

        @include wider-than($tablet-max) {
            margin-bottom: 5rem;
        }
    }

    .blurb-body {
        @include set-font(body-large);

        max-width: $text-content-max;
    }

    .webinars {
        align-items: center;
        display: grid;
        grid-gap: 1rem;
        grid-template-columns: auto 1fr;
        text-decoration: none;

        .icon-cell {
            align-items: center;
            border-radius: 50%;
            display: flex;
            font-size: 0.55 * $icon-size;
            height: $icon-size;
            justify-content: center;
            width: $icon-size;
        }

        h2 {
            @include set-font(h4);
        }
    }

    .ally-blurb {
        align-items: flex-start;
        display: grid;
        font-weight: inherit;
        grid-gap: $normal-margin;
        grid-template-columns: 1fr auto;

        > * {
            grid-column: 1;
        }

        .ally-logo {
            grid-column: 2;
            grid-row: 1 / span 2;

            > img {
                margin-top: 0.5rem;
                max-height: 6.8rem;
            }
        }
    }

    .ally-box {
        padding: 0;
        text-decoration: none;

        &:not(:hover) {
            color: inherit;
        }

        .top {
            align-items: center;
            display: flex;
            height: 100%;
            justify-content: center;
            padding: 4rem 0;
            position: relative;
            width: 100%;

            @include wider-than($tablet-max) {
                &:hover {
                    .hover-blurb {
                        opacity: 1;
                    }

                    img {
                        opacity: 0;
                    }
                }
            }

            .hover-blurb {
                left: $normal-margin;
                opacity: 0;
                position: absolute;
                top: 0;
                transition: opacity 0.3s ease-in-out;
                width: calc(100% - #{2 * $normal-margin});
            }

            img {
                transition: opacity 0.3s ease-in-out;
            }
        }

        .bottom {
            align-items: baseline;
            background-color: ui-color(default);
            border-bottom-left-radius: 0.4rem;
            border-bottom-right-radius: 0.4rem;
            display: flex;
            justify-content: space-between;
            padding: $normal-margin 2rem;
        }

        img {
            margin: 0;
            max-width: calc(100% - #{2 * $normal-margin});
            width: 20rem;
        }
    }
}
