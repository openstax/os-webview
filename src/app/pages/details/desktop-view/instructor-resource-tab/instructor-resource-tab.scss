@import 'pattern-library/core/pattern-library/headers';

$icon-size: 6rem;

.instructor-resources {
    display: grid;
    grid-gap: 4rem;
    margin-bottom: 3.5rem;

    h2 {
        font-weight: normal;
        margin: 0;
    }

    .free-stuff-blurb {
        display: grid;
        grid-gap: $normal-margin;
        grid-template-columns: auto;
        margin-bottom: 4rem;

        @include wider-than($tablet-max) {
            margin-bottom: 5rem;
        }
    }

    .divider {
        align-items: center;
        color: text-color(helper);
        display: grid;
        font-variant: small-caps;
        font-weight: bold;
        grid-column-gap: 2rem;
        grid-template-columns: 1fr auto 1fr;
        justify-items: center;
        margin-bottom: 3rem;
        margin-top: 3rem;
        text-transform: lowercase;

        .line {
            border: thin solid ui-color(form-border);
            width: 100%;
        }
    }

    .cards {
        align-content: start;
        display: grid;
        grid-column-gap: 3rem;

        &:not(.include-partners) {
            .resources {
                grid-template-columns: 1fr 1fr 1fr;
            }

            .partners {
                display: none;
            }
       }

        &.include-partners {
            @include width-between($phone-max, $tablet-max) {
                grid-template-columns: 1fr 1fr;

                .double {
                    grid-column: span 1;
                }
            }

            @include wider-than($tablet-max) {
                grid-template-columns: 2fr 1fr;
            }
        }

        .resources {
            grid-auto-rows: min-content;
            margin-top: 0;
        }
    }

    .blurb-body {
        @include set-font(body-large);

        max-width: $text-content-max;
    }

    .webinars {
        align-items: center;
        display: grid;
        grid-gap: 1rem;
        grid-template-columns: $icon-size 1fr;
        max-width: 100%;
        text-decoration: none;

        .icon-cell {
            align-items: center;
            border-radius: 50%;
            display: flex;
            font-size: 0.55 * $icon-size;
            height: $icon-size;
            justify-content: center;
            width: $icon-size;
        }

        .blurb {
            max-width: $text-content-max;
        }

        h2 {
            @include set-font(h4);
        }
    }
}
