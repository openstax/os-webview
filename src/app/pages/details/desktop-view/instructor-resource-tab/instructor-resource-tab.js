import React, {useState, useEffect} from 'react';
import FeaturedResources from '../../common/featured-resources/featured-resources.js';
import {instructorResourceBoxPermissions} from '../../common/resource-box/resource-box';
import ResourceBoxes, {VideoResourceBoxes} from '../../common/resource-box/resource-boxes';
import shuffle from 'lodash/shuffle';
import Partners from './partners/partners';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {RawHTML} from '~/components/jsx-helpers/jsx-helpers.jsx';
import {useUserStatus, usePartnerFeatures} from '../../common/hooks';
import './instructor-resource-tab.css';

function FreeStuff({heading, blurb}) {
    return (
        <div className="free-stuff-blurb">
            <RawHTML Tag="h2" html={heading} />
            <RawHTML className="blurb-body" html={blurb} />
        </div>
    );
}

function Webinar({url, text, blurb}) {
    return (
        <a href={url} className="webinars">
            <div className="icon-cell">
                <FontAwesomeIcon icon="desktop" />
            </div>
            <div className="blurb">
                <h2>{text}</h2>
                <RawHTML className="blurb" html={blurb} />
            </div>
        </a>
    );
}

function FeaturedResourcesSection({header, models}) {
    return (
        <div>
            <div className="featured-resources">
                <FeaturedResources headline={header} resources={models} />
            </div>
            <div className="divider">
                <div className="line"></div>
                see additional resources below
                <div className="line"></div>
            </div>
        </div>
    );
}

function resourceBoxModel(resourceData, userStatus, bookModel) {
    return Object.assign(
        {
            heading: resourceData.resourceHeading,
            description: resourceData.resourceDescription,
            creatorFest: resourceData.creatorFestResource,
            comingSoon: Boolean(resourceData.comingSoonText),
            comingSoonText: resourceData.comingSoonText,
            featured: resourceData.featured,
            k12: resourceData.k12,
            videoReferenceNumber: resourceData.videoReferenceNumber,
            trackResource: Boolean(userStatus.isInstructor) &&
                {
                    book: bookModel.id,
                    // eslint-disable-next-line camelcase
                    account_id: userStatus.userInfo.accounts_id,
                    // eslint-disable-next-line camelcase
                    resource_name: resourceData.resourceHeading
                },
            printLink: resourceData.printLink,
            bookModel
        },
        instructorResourceBoxPermissions(resourceData, userStatus, 'Instructor resources')
    );
}

function InstructorResourceTab({model, userStatus}) {
    const bookAbbreviation = model.salesforceAbbreviation;
    const featuredModels = model.bookFacultyResources
        .filter((r) => r.featured)
        .map((res) => resourceBoxModel(res, userStatus, model));
    const blogLinkModels = model.bookFacultyResources
        .filter((r) => r.linkText === 'View resources')
        .map((res) => resourceBoxModel(res, userStatus, model));
    const referenceModels = model.bookFacultyResources
        .filter((r) => r.videoReferenceNumber !== null)
        .map((res) => resourceBoxModel(res, userStatus, model));
    const otherModels = model.bookFacultyResources
        .filter((r) =>
            !r.featured && r.videoReferenceNumber === null &&
            r.linkText !== 'View resources'
        )
        .map((res) => resourceBoxModel(res, userStatus, model));

    const freeStuffContent = model.freeStuffInstructor.content;
    const freeStuff = {
        heading: freeStuffContent.heading,
        blurb: freeStuffContent.content,
        loggedInBlurb: freeStuffContent.contentLoggedIn
    };
    const webinarContent = model.webinarContent.content;
    const webinar = {
        text: webinarContent.heading,
        url: model.webinarContent.link,
        blurb: webinarContent.content
    };
    const communityResource = {
        heading: model.communityResourceHeading,
        logoUrl: model.communityResourceLogoUrl,
        url: model.communityResourceUrl,
        cta: model.communityResourceCta,
        blurb: model.communityResourceBlurb,
        featureUrl: model.communityResourceFeatureLinkUrl,
        featureText: model.communityResourceFeatureText
    };
    const partnerListLabel = model.partnerListLabel || '[partner_list_label]';
    const seeMoreText = model.partnerPageLinkText || '[partner_page_link_text]';
    const [blurbs, includePartners] = usePartnerFeatures(bookAbbreviation);
    const partnersModel = {
        title: partnerListLabel,
        seeMoreText,
        blurbs,
        badgeImage: '/images/partners/verified-badge.svg'
    };

    return (
        <div className="instructor-resources">
            <div>
                <FreeStuff {...freeStuff} />
                {
                    featuredModels.length > 0 &&
                        <FeaturedResourcesSection
                            header={model.featuredResourcesHeader}
                            models={featuredModels}
                        />
                }
                <div className={`cards ${includePartners}`}>
                    <div className="resources">
                        <VideoResourceBoxes
                            models={model.bookVideoFacultyResources}
                            blogLinkModels={blogLinkModels}
                            referenceModels={referenceModels}
                        />
                        <ResourceBoxes communityResource={communityResource} models={otherModels} />
                    </div>
                    <Partners
                        bookAbbreviation={bookAbbreviation}
                        model={partnersModel}
                    />
                </div>
            </div>
            <Webinar {...webinar} />
        </div>
    );
}

export default function LoadUserStatusThenInstructorTab({model}) {
    const userStatus = useUserStatus();

    if (!userStatus) {
        return null;
    }
    return (
        <InstructorResourceTab model={model} userStatus={userStatus} />
    );
}
