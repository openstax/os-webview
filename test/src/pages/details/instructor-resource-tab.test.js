import {InstructorResourceTabJsx} from '~/pages/details/instructor-resource-tab/instructor-resource-tab.jsx';
import details from '../../data/details-biology-2e.js';
import {transformData} from '~/helpers/controller/cms-mixin';
import {makeMountRender, snapshotify} from '../../../helpers/jsx-test-utils.jsx';

const instructor = {
    isStudent: false,
    isInstructor: true,
    userInfo: {
        id: 2
    }
};
const pageData = transformData(details);

describe('Instructor Resources Tab', () => {
    it('matches snapshot', () => {
        const wrapper = makeMountRender(InstructorResourceTabJsx, {
            bookAbbreviation: 'BIOLOGY',
            userStatusPromise: Promise.resolve(instructor),
            model: {
                resources: pageData.book_faculty_resources,
                allies: pageData.book_allies,
                freeStuff: {
                    heading: pageData.free_stuff_instructor.content.heading,
                    blurb: pageData.free_stuff_instructor.content.content,
                    loggedInBlurb: pageData.free_stuff_instructor.content.content_logged_in
                },
                webinar: {
                    text: pageData.webinar_content.content.heading,
                    url: pageData.webinar_content.link,
                    blurb: pageData.webinar_content.content.content
                },
                featuredResources: [],
                otherResources: [],
                communityResource: {
                    heading: pageData.community_resource_heading,
                    logoUrl: pageData.community_resource_logo_url,
                    url: pageData.community_resource_url,
                    cta: pageData.community_resource_cta,
                    blurb: pageData.community_resource_blurb,
                    featureUrl: pageData.community_resource_feature_link_url,
                    featureText: pageData.community_resource_feature_text
                }
            },
            dialogProps: {},
            partnerListLabel: 'Tech Partners',
            seeMoreText: 'See them all'
        })();

        expect(snapshotify(wrapper)).toMatchSnapshot();
    });
});
