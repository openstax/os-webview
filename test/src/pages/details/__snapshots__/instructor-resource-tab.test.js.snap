// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructor Resources Tab matches snapshot 1`] = `"<div class=\\"instructor-resources\\"><div><div class=\\"free-stuff-blurb\\"><h2><b>Academic freedom.</b> No catch.</h2><div class=\\"blurb-body\\"><p>Access free resources integrated with your book. <a href=\\"https://accounts.openstax.org/signup\\">Sign up</a> for an account to access locked content.</p></div></div><div class=\\"cards \\"><div class=\\"resources\\"><div class=\\"resource-box double\\"><div class=\\"top-line\\"><h3>OER Commons Hub Resources</h3><img src=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/documents/OERC_logo.jpg\\" alt=\\"\\"></div><div class=\\"blurb\\"><span>Members of the OpenStax community can join the Hub to share resources they've created to use with our biology book, like the interactive activity below. Find more resources and submit your own.</span><br><a href=\\"https://www.oercommons.org/groups/openstax-biology/1052/?__hub_id=27\\">Visit the Hub <i class=\\"fa fa-external-link-alt\\"></i></a></div><div class=\\"bottom\\"><div class=\\"left\\"></div><div class=\\"right\\"><a href=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/documents/Activities_for_engaging_students_in_Biology_using_animations_1.pdf\\">Biology animation activity</a></div></div></div></div><div class=\\"partners\\"><div class=\\"callout\\"></div><div class=\\"title-bar\\">Tech Partners<a class=\\"filter-for-book\\" href=\\"/partners\\">See them all</a></div><div class=\\"blurb-scroller\\"><div class=\\"blurbs\\"></div></div></div></div></div><a href=\\"https://openstax.org/webinars\\" class=\\"webinars\\"><div class=\\"icon-cell\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"desktop\\" class=\\"svg-inline--fa fa-desktop fa-w-18 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 576 512\\"><path fill=\\"currentColor\\" d=\\"M528 0H48C21.5 0 0 21.5 0 48v320c0 26.5 21.5 48 48 48h192l-16 48h-72c-13.3 0-24 10.7-24 24s10.7 24 24 24h272c13.3 0 24-10.7 24-24s-10.7-24-24-24h-72l-16-48h192c26.5 0 48-21.5 48-48V48c0-26.5-21.5-48-48-48zm-16 352H64V64h448v288z\\"></path></svg></div><div class=\\"blurb\\"><h2>Find a webinar</h2><div class=\\"blurb\\">Learn more by attending one of our free webinars featuring OpenStax authors, staff, and partners.</div></div></a></div>"`;
